pkg/validate/security_context.go

```diff
@@ -450,8 +502,123 @@ var _ = framework.KubeDescribe("Security Context", func() {
 			By("verify seccomp profile")
 			verifySeccomp(rc, containerID, []string{"grep", "ecc", "/proc/self/status"}, false, "0") // seccomp disabled
 		})
-	})
 
+		It("runtime should support an seccomp profile that blocks setting hostname with no sysCaps", func() {
+			privileged := false
+			expectContainerCreateToPass := true
+			By("create pod")
+			podID, podConfig = framework.CreatePodSandboxForContainer(rc)
+			By("create container with seccompBlockHostNameProfile and test")
+			containerID := createSeccompContainer(rc, ic, podID, podConfig,
+				"container-with-block-hostname-seccomp-profile-test-",
+				localhost+blockHostNameProfilePath, nil, privileged, expectContainerCreateToPass)
+			startContainer(rc, containerID)
+			Eventually(func() runtimeapi.ContainerState {
+				return getContainerStatus(rc, containerID).State
+			}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+			checkSetHostname(rc, containerID, false)
+		})
+
+		It("runtime should not support a custom seccomp profile without using localhost/ as a prefix", func() {
+			privileged := false
+			expectContainerCreateToPass := false
+			By("create pod")
+			podID, podConfig = framework.CreatePodSandboxForContainer(rc)
+			By("create container with seccompBlockHostNameProfile and test")
+			_ = createSeccompContainer(rc, ic, podID, podConfig,
+				"container-with-block-hostname-seccomp-profile-test-",
+				blockHostNameProfilePath, nil, privileged, expectContainerCreateToPass)
+		})
+
+		It("runtime should ignore a seccomp profile that blocks setting hostname when privileged", func() {
+			privileged := true
+			expectContainerCreateToPass := true
+			By("create privileged pod")
+			podID, podConfig = createPrivilegedPodSandbox(rc, true)
+			By("create privileged container with seccompBlockHostNameProfile and test")
+			containerID := createSeccompContainer(rc, ic, podID, podConfig,
+				"container-with-block-hostname-seccomp-profile-test-",
+				localhost+blockHostNameProfilePath, nil, privileged, expectContainerCreateToPass)
+			startContainer(rc, containerID)
+			Eventually(func() runtimeapi.ContainerState {
+				return getContainerStatus(rc, containerID).State
+			}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+			checkSetHostname(rc, containerID, true)
+		})
+
+		It("runtime should block setting hostname with no seccomp profile specified", func() {
```

Comment:
no seccomp profile means "unconfined", why set hostname will be blocked? By whom?
Default capabilities? If so, I feel like this is more a capability test instead of seccomp test.

Reply:
WE can add a test to create a default container, and check whether seccomp is enabled. That is useful.

Reply:
This is a duplicate test, I'll delete it.